src/library_browser.js

```diff
@@ -1042,34 +1045,35 @@ var LibraryBrowser = {
       };
       Browser.mainLoop.method = 'rAF';
     } else if (mode == 2 /*EM_TIMING_SETIMMEDIATE*/) {
-      if (!window['setImmediate']) {
+      if (typeof setImmediate === 'undefined') {
         // Emulate setImmediate. (note: not a complete polyfill, we don't emulate clearImmediate() to keep code size to minimum, since not needed)
         var setImmediates = [];
         var emscriptenMainLoopMessageId = 'setimmediate';
         function Browser_setImmediate_messageHandler(event) {
-          if (event.source === window && event.data === emscriptenMainLoopMessageId) {
+          if (event.data === emscriptenMainLoopMessageId || event.data.target === emscriptenMainLoopMessageId) {
```

Comment:
overall lgtm, nice work. just one note here, please add a comment for where the two cases come from. I assume one is from the normal case and one is from a pthread message?

Reply:
Yeah, that's right. I use the `{target: ... }` structure so that this is compatible with `--proxy-to-worker` mode.